src/webots/gui/WbSingleTaskApplication.cpp

```diff
@@ -44,13 +48,91 @@ void WbSingleTaskApplication::run() {
   else if (mTask == WbGuiApplication::VERSION)
     cout << tr("Webots version: %1").arg(WbApplicationInfo::version().toString(true, false, true)).toUtf8().constData() << endl;
   else if (mTask == WbGuiApplication::UPDATE_PROTO_CACHE)
-    updateProtoCacheFiles(mTaskArgument);
+    updateProtoCacheFiles();
   else if (mTask == WbGuiApplication::UPDATE_WORLD)
     WbWorld::instance()->save();
+  else if (mTask == WbGuiApplication::CONVERT) {
+    convertProto();
+  }
 
   emit finished(mTask == WbGuiApplication::FAILURE ? EXIT_FAILURE : EXIT_SUCCESS);
 }
 
+void WbSingleTaskApplication::convertProto() const {
+  QCommandLineParser cliParser;
+  cliParser.setApplicationDescription("Convert a PROTO file to URDF, WBO, or WRL file");
+  cliParser.addHelpOption();
+  cliParser.addPositionalArgument("input", "Path to the input PROTO file.");
+  cliParser.addPositionalArgument("output", "Path to the output URDF, WBO, or WRL file.");
+  cliParser.addOption(QCommandLineOption("p", "Override default PROTO parameters.", "parameter=value"));
+  cliParser.process(mTaskArguments);
+  QStringList positionalArguments = cliParser.positionalArguments();
```

Comment:
```suggestion
  const QStringList positionalArguments = cliParser.positionalArguments();
```

Reply:
This cannot be const as the object is later mutated with `pair[1].replace(...`